media-proxy/src/libfabric_cq.c

```diff
@@ -50,49 +50,55 @@
 
 #define CQ_TIMEOUT (-1)
 
-void rdma_cq_set_wait_attr(struct fi_cq_attr cq_attr, enum cq_comp_method method,
+void rdma_cq_set_wait_attr(struct fi_cq_attr *cq_attr, enum cq_comp_method method,
                            struct fid_wait *waitset)
 {
     switch (method) {
     case RDMA_COMP_SREAD:
-        cq_attr.wait_obj = FI_WAIT_UNSPEC;
-        cq_attr.wait_cond = FI_CQ_COND_NONE;
+        cq_attr->wait_obj = FI_WAIT_UNSPEC;
+        cq_attr->wait_cond = FI_CQ_COND_NONE;
         break;
     case RDMA_COMP_WAITSET:
         assert(waitset);
-        cq_attr.wait_obj = FI_WAIT_SET;
-        cq_attr.wait_cond = FI_CQ_COND_NONE;
-        cq_attr.wait_set = waitset;
+        cq_attr->wait_obj = FI_WAIT_SET;
+        cq_attr->wait_cond = FI_CQ_COND_NONE;
+        cq_attr->wait_set = waitset;
         break;
     case RDMA_COMP_WAIT_FD:
-        cq_attr.wait_obj = FI_WAIT_FD;
-        cq_attr.wait_cond = FI_CQ_COND_NONE;
+        cq_attr->wait_obj = FI_WAIT_FD;
+        cq_attr->wait_cond = FI_CQ_COND_NONE;
         break;
     case RDMA_COMP_YIELD:
-        cq_attr.wait_obj = FI_WAIT_YIELD;
-        cq_attr.wait_cond = FI_CQ_COND_NONE;
+        cq_attr->wait_obj = FI_WAIT_YIELD;
+        cq_attr->wait_cond = FI_CQ_COND_NONE;
         break;
     default:
-        cq_attr.wait_obj = FI_WAIT_NONE;
+        cq_attr->wait_obj = FI_WAIT_NONE;
         break;
     }
 }
 
-/*
- * fi_cq_err_entry can be cast to any CQ entry format.
- */
 static int rdma_spin_for_comp(ep_ctx_t *ep_ctx, struct fid_cq *cq, uint64_t *cur, uint64_t total,
-                              int timeout)
+                              int timeout, struct fi_cq_err_entry *entries)
 {
+    uint64_t entries_num;
     struct fi_cq_err_entry comp;
     struct timespec a, b;
     int ret;
 
+    if (!cur)
+        return EINVAL;
```

Comment:
Please check all errors that you're return and make them negative.
```suggestion
        return -EINVAL;
```

Reply:
fixed